engine/src/main/java/org/terasology/engine/world/chunks/pipeline/ChunkProcessingPipeline.java

```diff
@@ -112,27 +115,22 @@ private void processNewChunks(List<Chunk> chunks) {
             chunkProcessingInfo.setChunk(chunk);
             chunkProcessingInfo.nextStage(stages);
             chunkProcessingInfo.makeChunkTask();
+            processing.add(chunkProcessingInfo);
             chunkProcessingInfoMap.put(position, chunkProcessingInfo);
         }
-
-        while (processChunkTasks()) {
-            continue;
-        }
```

Comment:
The problem with removing the loop is this situation: let's say the queue has chunk A first and then chunk B, and chunk B was the last chunk to be generated. The next task for chunk A needs chunk B to have finished its task, so when we pop that off the queue we add it to `defer` and put it back on at the end. Then chunk B runs its task, and now if we tried A again it would work, but we don't do that because it's not in a loop and there aren't any more chunks coming.

This would manifest in game as the chunk not appearing until the player moves a little bit to get new chunks to generate, but would only happen very rarely and with the farthest-away chunks, so it's not actually a very big deal.

Reply:
I move them to a defer list and then re-add them to the queue.